cluster-agent/controllers/util.go

```diff
@@ -133,13 +137,13 @@ func DeleteArgoCDApplication(ctx context.Context, appFromList appv1.Application,
 
 		backoff.Reset()
 
-		// Add up to X minutes (eg 2 minutes) to wait for the deletion
-		expirationTime = expirationTime.Add(WaitForArgoCDToPerformNonfinalizerDeleteTimeout)
+		// Wait 2 minutes from the current time for the application to delete, before reporting an error.
```

Comment:
I'm sure you have a good reason. Can you share some information why this is needed?

Reply:
@drpaneas, good question. Once the finalizer is removed from the Application, Kubernetes should naturally clean up (delete) the Application, so we wait for it to do so.